routing/routing_tool_core/compare_routes.cpp

```diff
@@ -0,0 +1,69 @@
+#include "routing/routing_tool_core/compare_routes.hpp"
+#include "routing/routing_tool_core/utils.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include "base/assert.hpp"
+#include "base/logging.hpp"
+
+#include <algorithm>
+
+using namespace std;
+
+namespace routing_tools
+{
+namespace metrics
+{
+struct BasedOnPoints
```

Comment:
What is based on points?

Reply:
It is but it's bit unconventional. If you are using namespace metrics, plain BasedOnPoints is not so obvious.
Up to you.